src/bft.rs

```diff
@@ -1,16 +1,16 @@
-use crate::{units::SignedUnit, Data, Hasher, NodeIndex};
+use crate::{units::UncheckedSignedUnit, Data, Hasher, NodeIndex};
 use codec::{Decode, Encode};
 
 #[derive(Debug, Encode, Decode)]
 pub(crate) struct ForkProof<H: Hasher, D: Data, Signature: Clone + Encode + Decode> {
-    pub(crate) u1: SignedUnit<H, D, Signature>,
-    pub(crate) u2: SignedUnit<H, D, Signature>,
+    pub(crate) u1: UncheckedSignedUnit<H, D, Signature>,
+    pub(crate) u2: UncheckedSignedUnit<H, D, Signature>,
```

Comment:
Semantically this is wrong – a `ForkProof` should refer to two properly signed units, not the unchecked variants.

Reply:
Yes, but... We want this type to implement Encode + Decode, which is unimplementable for a struct storing a reference to a keybox...

Reply:
Right, so it would require a separate `UncheckedForkProof` type... Probably not worth it, what do you think?

Reply:
Yes, I think this is reasonable to only have an "unchecked" variant